### PR TITLE
Extend Dashboard E2E tests with UI checks and make them use Data Seeder

### PR DESCRIPTION
**Pull request by ndrsszsz**
[iOS Canvas Student E2E test suite Bitrise run](https://app.bitrise.io/build/852d8386-ddd8-4c5f-9061-c8582c3451a7)
- Added accessibilityIdentifier to several elements
- Changed pull_to_refresh parameter name to shouldPullToRefresh in AnnouncementHelper (as requested)
- Created DashboardHelper with some helper functions
- Upgraded all of the Dashboard tests to use DataSeeder and added plus UI visibility checks
- Upgraded Enrollment part of DataSeeder class to return an object containing the Enrollment ID
- Upgraded functions of Enrollment extension of DataSeeder class with discardableResult decorator (because currently only 1 test case makes use of the returned value)

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Please add only one release note.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>Please add only one affects.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This pull request will not generate a release note.</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>One or more files are below the minimum test coverage 50%</td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student

[MBL-16817](https://instructure.atlassian.net/browse/MBL-16817)
[MBL-16827](https://instructure.atlassian.net/browse/MBL-16827)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 92.01% | 92.01% | 0%
Core/Core/CourseSync/CourseSyncProgress/Model/CourseSyncProgressInteractor.swift | 0% | 0% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 2c81c632d1f1acdadf536badc6440c90ca7424ec
</p>

File: Student/StudentE2ETests/Dashboard/DashboardTests.swift [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    }
+
+    func testDashboardEditButtonDisplaysCorrectCourses() {
+        // MARK: Seed the usual stuff with 7 courses and student enrolled in them with all 7 different states
+        let student = seeder.createUser()
+        let courses = DashboardHelper.createCourses(number: 7)
+        seeder.enrollStudent(student, in: courses[0], state: .active)
+        seeder.enrollStudent(student, in: courses[1], state: .invited)
+        seeder.enrollStudent(student, in: courses[2], state: .completed)
+        seeder.enrollStudent(student, in: courses[3], state: .creation_pending)
```
Comment (idevainstr) [inline Student/StudentE2ETests/Dashboard/DashboardTests.swift:130]: Is there a way to test this one?

**Review by idevainstr — CHANGES_REQUESTED** [review]

File: Student/StudentE2ETests/Dashboard/DashboardTests.swift (hunk 1 above)
Comment (ndrsszsz) [reply Student/StudentE2ETests/Dashboard/DashboardTests.swift:130]: These lines are inside a test case. And there are multiple courses here. Which course do you mean of these 4 that you highlighted? This particular test is for checking the visibility of every course on the edit screen of the dashboard, depending on the enrollment state of each. What step do you miss that should be tested here?

File: Student/StudentE2ETests/Dashboard/DashboardTests.swift [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        if !showGradeToggle.isSelected {
+            showGradeToggle.tap()
+        }
+
+        // MARK: Tap Done button then check visibility of course again
+        var doneButton = DashboardHelper.doneButton.waitToExist()
+        XCTAssertTrue(doneButton.isVisible)
+
+        doneButton.tap()
+        pullToRefresh()
```
[the remark this answers is not in the document]
Comment (ndrsszsz) [reply Student/StudentE2ETests/Dashboard/DashboardTests.swift]: Fair enough. Done. I removed it and the test still passes. 👍

**Review by vargaat — APPROVED** [review]

**Review by idevainstr — APPROVED** [review]

**Pull request merged**